### PR TITLE
 Add time and distance thresholds for interpolation

### PR DESCRIPTION
**Pull request by andyphelps**
* Two new command line options added for spark job
* Add defaults for time and distance thresholds of 6hrs and 30km respectively

### CODE REVIEW

File: spark-job/src/main/scala/uk/gov/ukho/ais/rasters/Config.scala [hunk 1]
```diff
@@ -38,6 +42,26 @@ object Config {
           case resolution if resolution > 0 => success
           case _                            => failure("")
         },
+      opt[Long]('t', "timeThreshold")
+        .optional()
+        .valueName("<milliseconds>")
+        .text("threshold of time between pings in which interpolation will not occur")
+        .action((value, config) =>
+          config.copy(interpolationTimeThresholdMilliseconds = value))
+        .validate {
```
Comment (JoelClemence) [inline spark-job/src/main/scala/uk/gov/ukho/ais/rasters/Config.scala]: Can this logic be extracted as this is the same for distance and time?

File: spark-job/src/test/scala/uk/gov/ukho/ais/rasters/ComponentTest.scala [hunk 2]
```diff
@@ -101,6 +103,43 @@ class ComponentTest {
     assertPngBeenCreated()
   }
 
+  @Test
+  def whenAGappySampleIsInputtedWithGivenTimeThresholdThenATifIsProducedWithInterpolatedPoints()
```
Comment (JoelClemence) [inline spark-job/src/test/scala/uk/gov/ukho/ais/rasters/ComponentTest.scala]: Maybe rename other gappy test to state it is using default thresholds?

File: spark-job/src/test/scala/uk/gov/ukho/ais/rasters/InterpolatorTest.scala [hunk 3]
```diff
@@ -4,13 +4,17 @@ import java.util.Comparator
 
 import org.apache.commons.math3.util.Precision
 import org.assertj.core.api.Assertions.assertThat
-import org.junit.Test
+import org.junit.{Before, Test}
 
 import scala.collection.JavaConverters._
 
 class InterpolatorTest {
```
Comment (JoelClemence) [inline spark-job/src/test/scala/uk/gov/ukho/ais/rasters/InterpolatorTest.scala]: Does this need a unit test for the new functionality? Probably not but asking anyway?

File: spark-job/src/main/scala/uk/gov/ukho/ais/rasters/Interpolator.scala [hunk 4]
```diff
@@ -47,31 +45,41 @@ object Interpolator {
...
               prevPing.latitude,
               currPing.longitude,
-              currPing.latitude) <= DISTANCE_THRESHOLD
+              currPing.latitude) <= distanceThreshold
   }
 
   private def isTimeBetweenPingsWithinTheshold(prevPing: ShipPing,
-                                               currPing: ShipPing): Boolean = {
-    currPing.acquisitionTime - prevPing.acquisitionTime <= TIME_STEP_THRESHOLD
+                                               currPing: ShipPing,
```
Comment (oliverCausley) [inline spark-job/src/main/scala/uk/gov/ukho/ais/rasters/Interpolator.scala]: Threshold is spelt wrong for the method isTimeBetweenPingsWithinTheshold

File: spark-job/src/main/scala/uk/gov/ukho/ais/rasters/Config.scala (hunk 1 above)
Comment (eganjs) [reply spark-job/src/main/scala/uk/gov/ukho/ais/rasters/Config.scala]: I agree, code snippet for clarity:
```
       .validate {
          case threshold if threshold > 0 => success
          case _                          => failure("")
        },
```

Comment (JoelClemence) [reply spark-job/src/main/scala/uk/gov/ukho/ais/rasters/Config.scala]: There are also no tests for this validation

File: spark-job/src/test/scala/uk/gov/ukho/ais/rasters/InterpolatorTest.scala (hunk 3 above)
Comment (andyphelps) [reply spark-job/src/test/scala/uk/gov/ukho/ais/rasters/InterpolatorTest.scala]: No as it's being tested with a passed in value.

File: spark-job/src/main/scala/uk/gov/ukho/ais/rasters/Interpolator.scala (hunk 4 above)
Comment (andyphelps) [reply spark-job/src/main/scala/uk/gov/ukho/ais/rasters/Interpolator.scala]: Done

**Pull request merged**